src/NumericalAlgorithms/Interpolation/LinearInterpolationND.hpp

```diff
@@ -0,0 +1,374 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <memory>
+
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+
+namespace intrp {
+
+template <size_t Dimension, size_t NumberOfVariables, bool UniformSpacing>
+class LinearSpanInterpolationND {
```

Comment:
What do you think of naming this `MultiLinearSpanInterpolation`? The `ND` keeps confusing me with my initials 😅

Reply:
I didn't even think of that meaning of ND :)